### DISCUSSION TITLE
Dahomey.Json retirement

### DISCUSSION THREAD

**Discussion by mcatanzariti** [dahomey-technologies/Dahomey.Json#116] [Polls]
Hi,

When I decided to create Dahomey.Json, I wanted to extend System.Text.Json because im my opinion, it was missing some core features:
* Polymorphism
* Reference handling
* Contract customization

As of today, these features have been added to System.Text.Json in .net6 or .net7

Moreover additional features, that are not supported by Dahomey.Json, have been added to System.Text.Json like the source generator and it would require much effort to support it.

The way System.Text.Json evolves with feature overlapping with Dahomey.Json makes the exercise to support everything harder at each .net new release.

I'm strongly considering to discontinue Dahomey.Json but I would like to have your opinion in this survey.

Please, feel free to add any comment, beyond the survey !

Thank you,

Michaël

**Comment by MoazAlkharfan** [comment]
I would say as long as the features are not supported for .net 6 then it would be great to keep the library support for .net 6 only for these specific features.

- Polymorphism
- Reference handling
- Contract customization

**Comment by m-abs** [comment]
We rely heavily on `ShouldSerialize<PropertyName>()` in our project, I'm not sure how to achieve the same with `System.Text.Json`?
